src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PATH;
+
+import java.io.IOException;
+import java.util.Observable;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import seedu.address.MainApp;
+import seedu.address.commons.core.Config;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.util.StringUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+import seedu.address.storage.AddressBookStorage;
+import seedu.address.storage.JsonUserPrefsStorage;
+import seedu.address.storage.Storage;
+import seedu.address.storage.StorageManager;
+import seedu.address.storage.UserPrefsStorage;
+import seedu.address.storage.XmlAddressBookStorage;
+
+/**
+ * Imports contacts from a different Addressbook file
+ */
+public class ImportCommand extends UndoableCommand {
```

Comment:
Why is 'import' undoable? The user can just delete the file to undo it?

Reply:
It is the class which is already present. It means the command is "Undo" and "Redo" able. So it can be undone and redone. I am just extending out of that command

Reply:
I mean why not 'extends command' rather than 'extends undoableCommand'. Is there a need for it to be undoable?